[ShiJiaAo] iP

ShiJiaAo opened the pull request:
## Kal, Your Calendar Pal
> "Don't forget stuff like I always do" - Jia-Ao Shi, _Kal Developer_

Never ~remember~ forget your tasks again! Kal is here to help. Kal is:

- helpful 😄
- easy to use 🎉
- supportive 🌈
- amazing 😭

**WHAT CAN KAL DO?**

- [x] Add, View, Edit, Delete tasks
- [x] Mark tasks as done
- [x] Add deadlines, event time periods
- [x] **Search** (like Google 🤔)
- [ ] Reminders (coming soon)

So, how do I use Kal _**NOW**_?

1. [Download Kal here](https://github.com/ShiJiaAo/ip/releases/tag/A-Jar)
2. Copy the jar file into an empty folder
3. Open a command window in that folder
4. Run the command java -jar ip.jar (i.e., run the command in the same folder as the jar file)
5. Learn the user commands (simple)
6. Let Kal do your calendar work for **YOU** 🎉
#
Wondering how Kal works? Here is the `Launcher` class!!
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Commit:
change Duke.java to match Level-1 requirements

Commit:
Add the ability to store whatever text entered by the user and display them back to the user when requested.

Commit:
add Task.java class

Commit:
add Level-3 requirements

Commit:
Add support for tracking todos, deadlines, events

Commit:
modify auto ui testing to fit current project ver

Commit:
handle errors

Commit:
delete old files

Commit:
add support for deleting tasks from the list

Commit:
add author

Commit:
update EXPECTED.TXT

Commit:
Merge branch 'master' into add-gradle-support

Commit:
Organise project into packages

Commit:
Refactor commands and tasks into classes

Commit:
Clean up comments

Commit:
Add ui class

Commit:
Add TaskList class to track tasks

Commit:
Add JavaDocs

ezeAng commented on src/main/java/Duke/Commands/Delete.java:
```diff
@@ -0,0 +1,39 @@
+public class Delete extends Command {
+    private final int index;
+    private Task removed;
+    private int newSize;
```
i like the clear naming, adding 'new' to show that the size var is to be changed

ezeAng commented on src/main/java/Duke/Commands/Parser.java:
```diff
@@ -0,0 +1,128 @@
+
+    private String[] getEventTimes() {
+        String[] commandArr = this.command.split(" ");
+        int curInd = 1;
```
might be good to name it currIndex

ezeAng commented on src/main/java/Duke/Commands/Parser.java:
```diff
@@ -0,0 +1,128 @@
+            try {
+                checkLength(description);
+            } catch (DukeException ex) {
+                throw ex;
```
Looks like you really didnt like your ex

ezeAng commented on src/main/java/Duke/Commands/Tasks/Deadline.java:
```diff
@@ -0,0 +1,18 @@
+    }
+
+    private String getTaskClass() {
+        return "D";
```
could put this into a private String final, and return this.className

ezeAng reviewed (commented):
Left some comments!

hingen commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,42 @@
+ * Welcome to Duke
+ */
+public class Duke {
+    public static void main(String[] args) {
```
Should there be a header comment for this method?

hingen commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,42 @@
+package Duke;
```
I've noticed that all of your packages are named in Pascal case. Perhaps consider changing package names to lower case?

hingen commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,42 @@
+/**
+ * @author Shi Jiaao
+ *
+ * Welcome to Duke
```
It seems that there are multiple locations where a punctuation was not included at the end of each description. Perhaps consider adding a punctuation at the end of each description?

hingen commented on src/main/java/Duke/DukeException/DukeException.java:
```diff
@@ -0,0 +1,17 @@
+ */
+public class DukeException extends Exception {
+    private String errorMsg;
+    public DukeException(String message) {
```
Perhaps consider adding a header comment for this method?

hingen commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,64 @@
+
+    /**
+     * Constructor
+     */
```
Could consider this header instead:
```java
Creates a TaskList object.
```

hingen commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,51 @@
+
+    public Ui() {
+        // empty
+    }
```
Can consider adding a header comment to  this method or maybe just remove the constructor?

hingen commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,51 @@
+
+    /**
+     * Prints a descriptive message of a command
+     * @param command The command to be described
```
Perhaps add an empty line between description and parameter section?

hingen commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,64 @@
+     * Adds a task to the list
+     *
+     * @param newTask The task to be added to the list
+     */
```
I like the concise and descriptive comments you've added to your methods.

hingen commented on src/main/java/Duke/Commands/Command.java:
```diff
@@ -0,0 +1,22 @@
+ * This class is an abstract class representing all commands
+ */
+public abstract class Command {
+    // message is the command input
```
Perhaps consider changing this to a single-line JavaDoc comment:
```java
/** The command input. */
```

hingen reviewed (commented):
Some small coding convention violations that were repeated in multiple areas in your code. 

Overall code looks pretty clean :)
Good job conforming to the 120 line length limit, line wrapping in your code looks really clean.

Commit:
Add Storage class to take care of storage operations

Commit:
Add Storage class for local file storage

Commit:
Add formatter for date of deadlines and events

Commit:
Merge branch 'branch-Level-7'

Merge branch-Level-7

Commit:
Storage.java: clean up comments

Commit:
Add support for date recognition

Commit:
Merge level 7 and 8 branches

Commit:
Add find class and distribute printing to the ListTasks and Find commands

Commit:
Add JUnit tests for Parser and Storage classes

Commit:
Add JUnit tests for Parser and Storage classes

Commit:
Rename packages to lowercase

Commit:
Add CheckStyle Support

Commit:
Change import style checks in CheckStyle

Commit:
Change import style checks in CheckStyle

Commit:
Find.java: add abstraction layer to execute method to adhere to single layer abstraction principle

Commit:
Add basic GUI

Commit:
Remove dead code

Commit:
Fix formatting

Commit:
Pass printing in Duke.java to Ui.java

Commit:
Add constants for error messages

Commit:
Add all commands into GUI

Commit:
Fix minor formatting issues

Commit:
Add GUI support for all program functionality

Commit:
Update DialogBox.fxml

Commit:
Merge branch 'branch-Level-10'

Implement GUI support

Commit:
Storage.java: added constants for magic strings and numbers

Magic strings and numbers are present in Storage.java.

Having these magic strings and numbers may cause confusion as they may
be difficult to interpret, especially to other people who may read or
work on the code in the future.

Defining these strings and numbers as constants makes them more
descriptive. This improves readability of the code.

Commit:
Add assertion statements

Commit:
Fix minor code style flaws

Commit:
Add periods to the end of JavaDoc statements

Commit:
Refine code quality

Commit:
Improve code according to Code Quality guidelines

Commit:
build.gradle: Change mainClassName to "duke.Launcher"

mainClassName in build.gradle does not match with the actual main class
in the current project. JAR files generated using the command
"./gradlew shadowJar" are therefore unable to find the main class.

Let's change mainClassName to "duke.Launcher" as it is the actual entry
point of our program.

This will allow JAR files built using gradle to run properly.

ShiJiaAo committed:
Merge pull request #2 from ShiJiaAo/branch-A-Assertions

Merge Branch-A-Assertions to master

Commit:
Merge branch 'master' into branch-A-CodeQuality

ShiJiaAo committed:
Merge pull request #3 from ShiJiaAo/branch-A-CodeQuality

Merge branch-A-CodeQuality

Commit:
Add 'main' method into Duke.java

Commit:
Ui.java: fix printCommandMessage output

Commit:
Add test case

Commit:
Add extension C-DetectDuplicates

Commit:
Add unique cat personality

Commit:
Edit code format to conform to checkstyle rules

ShiJiaAo committed:
Add user guide to README.md

ShiJiaAo committed:
Update README.md

ShiJiaAo committed:
Update README.md

ShiJiaAo committed:
Update README.md

ShiJiaAo committed:
Update README.md

ShiJiaAo committed:
Create _config.yml

ShiJiaAo committed:
Update _config.yml

ShiJiaAo committed:
Add credits for external theme

ShiJiaAo committed:
Uncapitalize "event"

Commit:
Give more personality to Kal

Commit:
Merge branch 'master' of https://github.com/ShiJiaAo/ip

Commit:
Add product screenshot

Commit:
Finalize files for submission

ShiJiaAo committed:
Update README.md

ShiJiaAo committed:
Update README.md

ShiJiaAo committed:
Update README.md

ShiJiaAo committed:
Update README.md

ShiJiaAo committed:
Update README.md

ShiJiaAo committed:
Update README.md

ShiJiaAo committed:
Update README.md

Commit:
Convert pictures into png

Commit:
Merge branch 'master' of https://github.com/ShiJiaAo/ip

Commit:
Change pictures into png
